hyp3_autorift/crop.py

```diff
@@ -0,0 +1,129 @@
+    }
+
+
+def crop_netcdf_product(netcdf_file: Path) -> Path:
+    """
+
+    Args:
+        netcdf_file:
+    """
+    with xr.open_dataset(netcdf_file) as ds:
+        # this will drop X/Y coordinates, so drop non-None values just to get X/Y extends
+        xy_ds = ds.where(ds.v.notnull(), drop=True)
+
+        x_values = xy_ds.x.values
+        grid_x_min, grid_x_max = x_values.min(), x_values.max()
+
+        y_values = xy_ds.y.values
+        grid_y_min, grid_y_max = y_values.min(), y_values.max()
+
+        # Based on X/Y extends, mask original dataset
+        mask_lon = (ds.x >= grid_x_min) & (ds.x <= grid_x_max)
+        mask_lat = (ds.y >= grid_y_min) & (ds.y <= grid_y_max)
+        mask = (mask_lon & mask_lat)
+
+        cropped_ds = ds.where(mask, drop=True)
+        cropped_ds = cropped_ds.load()
+
+        # Reset data for mapping and img_pair_info data variables as ds.where() extends data of all data variables
+        # to the dimensions of the "mask"
+        cropped_ds['mapping'] = ds['mapping']
+        cropped_ds['img_pair_info'] = ds['img_pair_info']
+
+        # Compute centroid longitude/latitude
+        center_x = (grid_x_min + grid_x_max) / 2
+        center_y = (grid_y_min + grid_y_max) / 2
+
+        # Convert to lon/lat coordinates
+        projection = ds['mapping'].attrs['spatial_epsg']
+        to_lon_lat_transformer = pyproj.Transformer.from_crs(
+            f"EPSG:{projection}",
+            'EPSG:4326',
+            always_xy=True
+        )
+
+        # Update centroid information for the granule
+        center_lon_lat = to_lon_lat_transformer.transform(center_x, center_y)
+
+        cropped_ds['mapping'].attrs['latitude'] = round(center_lon_lat[1], 2)
+        cropped_ds['img_pair_info'].attrs['longitude'] = round(center_lon_lat[0], 2)
+
+        # Update mapping.GeoTransform
+        x_cell = x_values[1] - x_values[0]
+        y_cell = y_values[1] - y_values[0]
+
+        # It was decided to keep all values in GeoTransform center-based
+        cropped_ds['mapping'].attrs['GeoTransform'] = f"{x_values[0]} {x_cell} 0 {y_values[0]} 0 {y_cell}"
+
+        # Compute chunking like AutoRIFT does:
+        # https://github.com/ASFHyP3/hyp3-autorift/blob/develop/hyp3_autorift/vend/netcdf_output.py#L410-L411
+        dims = cropped_ds.dims
+        chunk_lines = np.min([np.ceil(8192 / dims['y']) * 128, dims['y']])
+        two_dim_chunks_settings = (chunk_lines, dims['x'])
+
+        encoding = ENCODING_TEMPLATE.copy()
+        if not netcdf_file.name.startswith('S1'):
+            for radar_variable in ['M11', 'M12', 'va', 'vr']:
+                del encoding[radar_variable]
+
+        for _, attributes in encoding.items():
+            if attributes['_FillValue'] is not None:
+                attributes['chunksizes'] = two_dim_chunks_settings
+
+        cropped_file = netcdf_file.with_stem(f'{netcdf_file.stem}_cropped')
+        cropped_ds.to_netcdf(cropped_file, engine='h5netcdf', encoding=encoding)
```

Comment:
`engine='h5netcdf'` here adds the `h5netcdf` dependency. 

It's not strictly needed, but it does increase the performance of this function. 

Overall, I'm very much **not** happy with how long it takes to crop the netCDF file, and we should take another look at it. It's not blocking, but annoying enough to make an issue about if we don't address it in this PR.